### PR TITLE
Added support for node_modules shims

### PR DESCRIPTION
**Pull request by TiddoLangerak**
Successor of #6. Turned out that glob patterns weren't actually powerful enough to detect node_modules folders, so it's a bit more complicated than just that.

@Magnetme/developers - Ready for review!

### CODE REVIEW

**Comment by TiddoLangerak** [comment]
@Magnetme/developers - poke.

File: src/plugins/shim.js [hunk 1]
```diff
@@ -2,54 +2,134 @@ let through = require('through2');
...
+ */
+function extractPackageShims(shims) {
+	let packageShims = {};
+	Object.keys(shims)
+		.filter((shimKey) => !path.isAbsolute(shimKey))
+		//This second filter tries to filter out glob patterns by checking if the first character
+		//could be the first character of a glob pattern. These are all rarely used tokens in files
+		//and all not valid for npm packages, so it's safe to assume that we don't exclude valid package
+		//names here.
+		.filter((shimKey) => !shimKey.match(/^[\*\!\+\{\[\?]/))
```
Comment (wouterwillems10) [inline src/plugins/shim.js]: Can you give an example of what is matched here? Getting a headache by just looking at this regex :hamster:

**Comment by wouterwillems10** [comment]
I am confused, what is meant by a shim for a file?

File: src/plugins/shim.js (hunk 1 above)
Comment (TiddoLangerak) [inline src/plugins/shim.js]: See the comment above, glob patterns (e.g. `'**/foo`, `*.js`)

**Comment by TiddoLangerak** [comment]
A shim is a compatibility layer, so in this case a shim makes scripts that do not use the CommonJS module standard compatible with CommonJS scripts (and thus Yoloader)

**Comment by wouterwillems10** [comment]
Right, gotcha

File: src/plugins/shim.js (hunk 1 above)
Comment (wouterwillems10) [inline src/plugins/shim.js]: yeah i checked the comment but didnt really understand it fully. Thnx to the example, now i do

Comment (TiddoLangerak) [inline src/plugins/shim.js]: Ahh ok, will include the examples as well in code then

**Comment by wouterwillems10** [comment]
:+1:

**Pull request merged**